leftwm-core/src/models/workspace.rs

```diff
@@ -20,14 +19,19 @@ pub struct Workspace {
     pub xyhw: Xyhw,
     xyhw_avoided: Xyhw,
     pub max_window_width: Option<Size>,
+    /// Output (monitor) the workspace is linked to.
+    pub output: String,
+    /// ID of workspace on output. Starts with 1.
+    pub num: usize,
```

Comment:
I recently asked on the Discord:
> Is there a reason that workspace.id is an Option and is there a reason why it is a signed integer rather than unsigned?

And @VuiMuich hinted that you are actually touching this in this PR, so I'm looking forward to the merge :tada:

Reply:
I tried to differentiate against the old ids (especially for people upgrading from their old config). If you want me to change that I can of course do it.

Reply:
I'd prefer `id` just for consistency reasons, but maybe @VuiMuich @AethanFoot have opinions on it. I'm not very knowledgeable about the workspace configurations. Why'd one set the id manually? I suppose the id is only used to refer to workspaces in the commands right? Would it be enough if we just automatically number them $[1..n]$ in the order they were defined? Tag IDs are also assigned in the same way.

The reason I'm focused on the IDs specifically is because I'm currently working on integrating the layout-library, and I want to store the different layouts in a `Map<usize, Vec<LayoutDefinition>>` where the `usize` key stands for Tag ID or Workspace ID depending on the `LayoutMode`. When Tag ID and Workspace ID behave the same and are assigned deterministically it might simplify that part for me.

Reply:
btw, thanks for your patience and for keeping the PR up with the main branch, I know your PR has been open for a while :heart:

Reply:
Okay, I will rename `num` to `id` then. And yes, this is surely an interesting idea that should be discussed. I would be in favor of it as it would untangle some config things (as most people don't need multiple workspaces per monitor anyways). I would certainly ask @VuiMuich about removing ids from the config entirely and setting them automatically before implementing that.

Reply:
So would we strictly assign IDs based on its pixel coordinates, rows before columns?

If I recall it right, the idea of keeping the numerical id configurable was to be as less breaking as possible, but we will break for a couple of scenarios anyways, right?

Reply:
No, the idea is to assign ids in order of definition in the config. In combination with output names ass scenarios should be covered.

Reply:
> No, the idea is to assign ids in order of definition in the config. In combination with output names all scenarios should be covered.

That's actually quite good. If we discussed this earlier and  I din't get it I apologise sincerely.

Reply:
So I should do it?

Reply:
Yes, please.

Reply:
Okay. I have to admit I did not remember what I already wrote. So yes, workspace ids are **already** only automatically assigned. Seems like past me was intelligent enough to already figure this out. I know I had it implemented at one point so that it would be needed to be specified in the config but I already changed it as discussed. :sweat_smile: